keystore/v2/keystore/filesystem/keyStore.go

```diff
@@ -138,6 +139,34 @@ func (s *KeyStore) OpenKeyRingRW(path string) (api.MutableKeyRing, error) {
 	return ring, nil
 }
 
+// ListKeyRings enumerates all key rings present in this key store.
+func (s *KeyStore) ListKeyRings() (rings []string, err error) {
```

Comment:
can we be sure that we return only paths to rings? that in this folder wasn't created an extra file which not related to keystore?

Reply:
> can we be sure that we return only paths to rings?

Right now: not so much. If the user adds some extra files then we'll pick them up, try parsing them, and fail either when listing keys (due to unexpected naming), or later during export (due to mismatching content).

There are ideas for improved integrity – key directories – which are cryptographically signed lists of key rings stored in a key store. Once that is implemented we can rely on the key directories to list the key rings and ignore (or report) any extra files.

We also store some bookkeeping files for locking and versioning, those are filtered out.